packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -318,10 +404,8 @@ private MethodBodyStatement[] BuildPrimaryConstructorBody(IReadOnlyList<Paramete
             if (_authorizationHeaderConstant != null && _apiKeyAuthField != null)
```

Comment:
I think the condition needs to consider which constructor we are in rather than whether there is an apiKeyAuthField or not. Also, it seems we are missing the logic to add the bearer token policy.

Reply:
yeah this is the root cause of causing https://github.com/Azure/azure-sdk-for-net/pull/47528#discussion_r1882630322
Let me fix this

Reply:
I have fixed this and introduced some test cases here to validate the body of primary ctors (previous we are not doing this)